MdeModulePkg/Library/DeviceManagerUiLib/DeviceManager.c

```diff
@@ -557,194 +557,201 @@ CreateDeviceManagerForm (
+      FreePool (String);
+      FreePool (NewStringTitle);
+    }
   }
 
   //
   // Allocate space for creation of UpdateData Buffer
   //
   StartOpCodeHandle = HiiAllocateOpCodeHandle ();
   ASSERT (StartOpCodeHandle != NULL);
+  // MU_CHANGE - Verify StartOpCodeHandle and EndOpCodeHandle are valid before using
+  if (StartOpCodeHandle != NULL) {
+    EndOpCodeHandle = HiiAllocateOpCodeHandle ();
+    ASSERT (EndOpCodeHandle != NULL);
+    if (EndOpCodeHandle != NULL) {
+      //
+      // Create Hii Extend Label OpCode as the start opcode
+      //
+      StartLabel               = (EFI_IFR_GUID_LABEL *)HiiCreateGuidOpCode (StartOpCodeHandle, &gEfiIfrTianoGuid, NULL, sizeof (EFI_IFR_GUID_LABEL));
+      StartLabel->ExtendOpCode = EFI_IFR_EXTEND_OP_LABEL;
+      //
+      // According to the next show Form id(mNextShowFormId) to decide which form need to update.
+      //
+      StartLabel->Number = (UINT16)(LABEL_FORM_ID_OFFSET + NextShowFormId);
 
-  EndOpCodeHandle = HiiAllocateOpCodeHandle ();
-  ASSERT (EndOpCodeHandle != NULL);
-
-  //
-  // Create Hii Extend Label OpCode as the start opcode
-  //
-  StartLabel               = (EFI_IFR_GUID_LABEL *)HiiCreateGuidOpCode (StartOpCodeHandle, &gEfiIfrTianoGuid, NULL, sizeof (EFI_IFR_GUID_LABEL));
-  StartLabel->ExtendOpCode = EFI_IFR_EXTEND_OP_LABEL;
-  //
-  // According to the next show Form id(mNextShowFormId) to decide which form need to update.
-  //
-  StartLabel->Number = (UINT16)(LABEL_FORM_ID_OFFSET + NextShowFormId);
-
-  //
-  // Create Hii Extend Label OpCode as the end opcode
-  //
-  EndLabel               = (EFI_IFR_GUID_LABEL *)HiiCreateGuidOpCode (EndOpCodeHandle, &gEfiIfrTianoGuid, NULL, sizeof (EFI_IFR_GUID_LABEL));
-  EndLabel->ExtendOpCode = EFI_IFR_EXTEND_OP_LABEL;
-  EndLabel->Number       = LABEL_END;
-
-  //
-  // Get all the Hii handles
-  //
-  HiiHandles = HiiGetHiiHandles (NULL);
-  ASSERT (HiiHandles != NULL);
-
-  //
-  // Search for formset of each class type
-  //
-  for (Index = 0; HiiHandles[Index] != NULL; Index++) {
-    Status = HiiGetFormSetFromHiiHandle (HiiHandles[Index], &Buffer, &BufferSize);
-    if (EFI_ERROR (Status)) {
-      continue;
-    }
+      //
+      // Create Hii Extend Label OpCode as the end opcode
+      //
+      EndLabel               = (EFI_IFR_GUID_LABEL *)HiiCreateGuidOpCode (EndOpCodeHandle, &gEfiIfrTianoGuid, NULL, sizeof (EFI_IFR_GUID_LABEL));
+      EndLabel->ExtendOpCode = EFI_IFR_EXTEND_OP_LABEL;
+      EndLabel->Number       = LABEL_END;
 
-    Ptr = (UINT8 *)Buffer;
-    while (TempSize < BufferSize) {
-      TempSize += ((EFI_IFR_OP_HEADER *)Ptr)->Length;
-      if (((EFI_IFR_OP_HEADER *)Ptr)->Length <= OFFSET_OF (EFI_IFR_FORM_SET, Flags)) {
-        Ptr += ((EFI_IFR_OP_HEADER *)Ptr)->Length;
-        continue;
-      }
+      //
+      // Get all the Hii handles
+      //
+      HiiHandles = HiiGetHiiHandles (NULL);
+      ASSERT (HiiHandles != NULL);

```

Comment:
Should we ditch out if HiiHandles is NULL?

Reply:
I think it is safe enough to gate any of the code with the ifs.

This function is a void, so it is not like  we can bubble up an error status to the caller.